app/javascript/app/ssn-field.js

```diff
@@ -0,0 +1,47 @@
+import Cleave from 'cleave.js';
+
+const { I18n } = window.LoginGov;
+
+/* eslint-disable no-new */
+function formatSSNField() {
+  const inputs = document.querySelectorAll('input.ssn-toggle[type="password"]');
+
+  if (inputs) {
+    [].slice.call(inputs).forEach((input, i) => {
+      const el = `
+        <div class="mt1 right">
+          <label class="btn-border" for="ssn-toggle-${i}">
+            <div class="checkbox">
+              <input id="ssn-toggle-${i}" type="checkbox">
+              <span class="indicator"></span>
+              ${I18n.t('forms.ssn.show')}
+            </div>
+          </label>
+        </div>`;
+      input.insertAdjacentHTML('afterend', el);
+
+      let ssnCleave;
+      const toggle = document.getElementById(`ssn-toggle-${i}`);
+      toggle.addEventListener('change', function () {
+        input.type = toggle.checked ? 'text' : 'password';
+
+        if (!toggle.checked) {
```

Comment:
Nit: With `if` `else`, I find it's less confusing if the `if` condition isn't negated ("if [x] else [not x]" rather than "if [not x] else [not not x]").

```suggestion
        if (toggle.checked) {
```

(more changes necessary than suggestion indicates)

Reply:
Yeah +1 I'm a big fan of using the positive (not-negated) case first

Reply:
Done!